app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/InsuredRow.kt

```diff
@@ -0,0 +1,185 @@
+package com.hedvig.android.feature.editcoinsured.ui
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.icons.HedvigIcons
+import com.hedvig.android.core.icons.hedvig.normal.X
+import com.hedvig.android.core.icons.hedvig.small.hedvig.Lock
+import com.hedvig.android.core.icons.hedvig.small.hedvig.Plus
+import com.hedvig.android.core.ui.text.HorizontalItemsWithMaximumSpaceTaken
+import hedvig.resources.R
+
+@Composable
+internal fun InsuredRow(
+  displayName: String?,
+  details: String?,
+  hasMissingInfo: Boolean,
+  allowEdit: Boolean,
+  isMember: Boolean,
+  onRemove: () -> Unit,
+  onEdit: () -> Unit,
+) {
+  HorizontalItemsWithMaximumSpaceTaken(
+    startSlot = {
+      Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier.padding(vertical = 16.dp),
+      ) {
+        Column {
+          if (displayName != null) {
+            Text(
+              text = displayName,
+              color = if (isMember) {
+                MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.5f)
```

Comment:
If we want a "disabled for interaction" sort of look here, onSurfaceVariant already achieves this.
If we want an even more toned down version you can probably use onSurface with alpha of this  https://github.com/HedvigInsurance/android/blob/03cecb9b5c957d0f15084f67cbe0e49d7e102d7d/app/core/core-design-system/src/main/kotlin/com/hedvig/android/core/designsystem/material3/HedvigMaterial3ColorScheme.kt#L194 for the constant instead.

I'd like to say that just using `onSurfaceVariant` by itself is good enough for the "disabled text" situations, all other options become way too low-contrast with the background making the text downright unreadable for users with any sort of visual difficulties.
It's the exact same reason for this comment here 
<img width="1089" alt="image" src="https://github.com/HedvigInsurance/android/assets/44558292/6cffc4bf-b3f8-4c8a-8ebe-1dadb053f4a8">
and why sometimes I may look like I am using the wrong colors from design, but it's not by accident that I do so.

Reply:
Good info, cheers